noise-research/secure-transfer.md

```diff
@@ -0,0 +1,243 @@
+# Secure Transfers with Noise
+
+In this document we describe a compound protocol to enable two devices to mutually authenticate and securely exchange (arbitrary) information. 
+It consists of two main subprotocols or *phases*:
+
+- [Device Pairing](#Device-Pairing): the devices exchange and authenticate their long term device ID static keys;
+- [Secure Transfer](#Secure-Transfer): the devices securely exchange in encrypted form information using key material obtained during a successful pairing phase.
+
+## Device Pairing
+
+In the pairing phase, a device `B` requests to be paired to a device `A`. Once the two devices are paired, the devices will be mutually authenticated and will share a Noise session within which they can securely exchange information.
+
+The request is made by exposing a QR code that, by default, has to be scanned by device `A`. 
+If device `A` doesn't have a camera while device `B` does, [it is possible](#Rationale) to execute a slightly different pairing (with same security guarantees), where `A` is exposing a QR code instead.
+
+### Employed Cryptographic Primitives
+
+- `H`: the underlying hash function, i.e. SHA-256;
+- `HKDF`: a key derivation function (based on SHA-256);
+- `Curve25519`: the underlying elliptic curve for Diffie-Hellman (DH) operations.
+
+### The `WakuPairing` Noise Handshake
+
+The devices execute a custom handshake derived from `X1X1`, where they mutually exchange and authenticate their device static keys by exchanging messages over the content topic
+
+```
+contentTopic = /{application-name}/{application-version}/wakunoise/1/sessions-{shard-id}/proto
+```
+
+The handshake, detailed in next section, can be summarized as:
+
+```
+WakuPairing:
+0.   <- eB              {H(sB||r), contentTopicParams, messageNametag}
+     ...
+1.   -> eA, eAeB        {H(sA||s)}   [authcode]
+2.   <- sB, eAsB        {r}
+3.   -> sA, sAeB, sAsB  {s}
+
+{}: payload,    []: user interaction
+```
+
+### Protocol Flow
+
+1. The device `B` exposes through a QR code a Base64 serialization of:
```

Comment:
How do these numbered steps map to the summarized figure above? The numbers do not match 7 steps vs 3
Update: I figured out the relation, but it is very hard at the first glance. More explanation can clarify the relation between the two.